src/mechanics/constitutive/ModifiedMises.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+#include "base/Exception.h"
+#include "mechanics/constitutive/ConstitutivePlaneStateEnum.h"
+#include "mechanics/constitutive/EngineeringStrain.h"
+#include "mechanics/constitutive/EngineeringStrainInvariants.h"
+
+namespace NuTo
+{
+namespace Constitutive
+{
+/**
+ * Equivalent strain (strain norm) based on the modified mises norm.
+ * De Vree et al. "Comparision of nonlocal approaches in continuum damage mechanics", Computers \& Structures,
+ * 1995.
+ * https://doi.org/10.1016/0045-7949(94)00501-S
+ *
+ * It includes the value `k` in a way that expresses the ratio of the materials compressive to tensile strength:
+ * A
+ * uniaxial tensile strain and a k-times higher uniaxial compressive strain both lead to the same equivalent
+ * strain.
+ * With the strain tensor invariant I1 and the deviatoric strain tensor invariant J2, it reads
+ * \f[/
+ * \varepsilon_\text{eq}(\boldsymbol \varepsilon) = \frac{k-1}{2k(1-2\nu)}I_1 +
+ * \frac{1}{2k}\sqrt{\left(\frac{k-1}{1-2\nu}I_1\right)^2 + \frac{2k}{(1+\nu)^2}J_2}
+ * \f]
+ *
+ * @tparam TDim dimension 1,2,3
+ */
+template <int TDim>
+class ModifiedMises
```

Comment:
the naming should somehow include what is done (it is not a plasticity model with a modified Mises yield function, but just calculates the Mises - norm). Also, it should be somehow clear that is using the strains (and not the stresses) - or is there a possibility to reuse this when I'm converting the standard Mises-plasticity model to pde and have to define the yield function also as a function of J2?

Reply:
The voight notation makes it hard to reuse `strain` methods for `stress`. We can try to reuse `mechanics/constitutive/EngineeringStrainInvariants.h`. They contain a method `ToTensor` that is different for stresses and strains. The invariants itself only depend on the tensor then. Does that make sense?